lib/pbench/server/database/models/datasets.py

```diff
@@ -833,12 +848,16 @@ def validate(dataset: Optional[Dataset], key: str, value: Any) -> Any:
         should be used in most contexts where client-visible metadata paths
         are used.
 
-        1) For 'dataset.name', we require a non-empty UTF-8 encoded string.
-        2) For 'server.deletion', the string must be an ISO date/time string,
-           and we fail otherwise. We store only the UTC date, as we don't
-           guarantee that deletion will occur at any specific time of day. The
-           maximum retention period (from upload) is defined by the server
-           configuration dataset-lifetime property.
+        1) 'dataset.name': a non-empty UTF-8 encoded string.
+        2) 'server.deletion': an ISO date/time string. We store only the UTC
+           date, as we don't guarantee that deletion will occur at any specific
+           time of day. The maximum retention period (from upload) is defined
+           by the server configuration dataset-lifetime property.
```

Comment:
That's an interesting philosophical/architectural issue to take up separately. Do we want people to be able to "cheat" (or, more architecturally, is it actually cheating) to repeatedly advance the expiration time of a dataset? If we really want to allow this we could just allow disabling expiration entirely ... then again, everyone might just do that because it's easy and they "might want it" whereas requiring them to serially bump it by the threshold means that forgotten datasets will always be deleted eventually.

Recorded in [PBENCH-1078](https://issues.redhat.com/browse/PBENCH-1078)